src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.TaskType;
+import duke.task.ToDo;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs the storage.
+     *
+     * @param filePath the filepath of the storage file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Reads the data of tasks from previous session's file as TaskList.
+     * If there's no file yet, create an empty file, and return empty list.
+     *
+     * @return the list of tasks from previous session
+     */
+    public TaskList readFile() {
+        List<Task> taskList = new ArrayList<>();
+        try {
+            File file = new File(filePath);
+            if (!file.createNewFile()) {
+                File myObj = new File(filePath);
+                Scanner myReader = new Scanner(myObj);
+                while (myReader.hasNextLine()) {
+                    String[] data = myReader.nextLine().split(" \\| ");
+                    if (data.length == 0) {
+                        break;
+                    }
+                    switch (data[0]) {
+                    case "TODO":
+                        taskList.add(new ToDo(data[2], TaskType.TODO,
+                                data[1].equals("1")));
+                        break;
+                    case "DEADLINE":
+                        taskList.add(new Deadline(data[2], TaskType.DEADLINE,
+                                LocalDate.parse(data[3]), data[1].equals("1")));
+                        break;
+                    case "EVENT":
+                        taskList.add(new Event(data[2], TaskType.EVENT,
+                                LocalDate.parse(data[3]), data[1].equals("1")));
+                        break;
+                    default:
+                        return new TaskList(taskList);
+                    }
+
+                }
+                myReader.close();
+            }
+
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        }
+        System.out.println("(Your tasks from previous session has been successfully loaded.)");
+        return new TaskList(taskList);
+    }
+
+    /**
+     * Updates the file with newest task data, write the updated tasks into the file.
+     *
+     * @param taskList the updated task list
+     */
+    public void updateFile(TaskList taskList) {
+        String updatedString = "";
+        for (Task t : taskList.getTaskList()) {
+            updatedString += t.getType();
+            updatedString += " | ";
```

Comment:
Perhaps extracting the data representation to specific task classes could make your code clearer.